hs_core/tasks.py

```diff
@@ -158,6 +167,63 @@ def nightly_periodic_task_check():
         file.write(timezone.now().strftime('%m/%d/%y %H:%M:%S'))
 
 
+@celery_app.task(ignore_result=True, base=HydroshareTask)
+def nightly_repair_resource_files():
+    """
+    Run repair_resource management command on resources updated in the last day
+    """
+    from hs_core.management.utils import check_time, repair_resource
+    start_time = time.time()
+    cuttoff_time = timezone.now() - timedelta(days=1)
+    recently_updated_resources = [res for res in BaseResource.objects.all() if res.last_updated >= cuttoff_time]
+    repaired_resources = []
+    try:
+        for res in recently_updated_resources:
+            check_time(start_time, settings.NIGHTLY_RESOURCE_REPAIR_DURATION)
+            repair_resource(res, logger)
+            repaired_resources.append(res)
+
+        # spend any remaining time fixing resources that weren't updated in the last day
+        for res in BaseResource.objects.exclude(short_id__in=recently_updated_resources):
```

Comment:
Realized that this might churn through the same subset of resources every night, checking them and running out of time before it can move on. Will ponder a better way